amplify/backend/function/jcProcessDMAlerts/src/index.js

```diff
@@ -238,24 +288,57 @@ async function Execute(event) {
         const content = record.dynamodb.NewImage.content.S
         const from = record.dynamodb.NewImage.userId.S
         const recipients = record.dynamodb.NewImage.recipients.L.map((item) => item.S)
-        console.log(recipients)
+        let recipientsToBeNotified = recipients
+        console.log({ recipients })
+        const lessonID = messageRoomID.split("-").slice(1, 6).join("-") // is assignment id
+        const courseLesson = await getCourseLesson(lessonID)
+        const courseWeek = await getCourseWeek(courseLesson.courseWeekID)
+        const courseInfo = await getCourseInfo(courseWeek.courseInfoID)
+        console.log({ courseInfo })
+        if (courseInfo?.separatedTriads) {
+          console.log("Filtering recipients.")
+          const backOfficeStaff =
+            courseInfo?.backOfficeStaff?.items?.map((item) => item?.userID) ?? []
+          console.log({ backOfficeStaff })
+          const instructors = courseInfo?.instructors?.items?.map((item) => item?.userID) ?? []
+          console.log({ instructors })
+          const triads = courseInfo?.triads?.items ?? []
+          const userTriad = triads?.find(
+            (triad) =>
+              triad?.coaches?.items?.find((item) => item?.userID === from) ||
+              triad?.users?.items?.find((item) => item?.userID === from)
+          )
+          console.log({ userTriad })
+          const coaches = userTriad?.coaches?.items?.map((item) => item?.userID) ?? []
+          const users = userTriad?.users?.items?.map((item) => item?.userID) ?? []
+          console.log({ coaches })
+          console.log({ users })
+          recipientsToBeNotified = recipients.filter(
+            (item) =>
+              coaches.includes(item) ||
+              users.includes(item) ||
+              instructors.includes(item) ||
+              backOfficeStaff.includes(item)
+          )
+        } else {
+          console.log("Skipped filtering recipients.")
+        }
+        recipientsToBeNotified = recipientsToBeNotified.filter((item) => item != from)
+        console.log({ recipientsToBeNotified })
```

Comment:
This is missing a check to see whether it is a DM or Course type.
It should skip this section if it is not Course type...
```
if(messageRoomID.startsWith("course")) { 
    
}